core/drivers/crypto/caam/acipher/caam_prime.c

```diff
@@ -719,12 +717,62 @@ static void do_checks_primes(uint32_t *desc, const struct caambuf *p,
 	RSA_DUMPDESC(desc);
 }
 
+/*
+ * Run the Primes descriptor.
+ *
+ * @desc   Descriptor built
+ * @prime  Prime generation data
+ */
+static enum caam_status run_primes(uint32_t *desc, struct prime_data *data)
+{
+	enum caam_status retstatus = CAAM_FAILURE;
+	struct caam_jobctx jobctx = { };
+
+	cache_operation(TEE_CACHEFLUSH, data->p->data, data->p->length);
+
+	if (data->q)
+		cache_operation(TEE_CACHEFLUSH, data->q->data, data->q->length);
+
+	jobctx.desc = desc;
+	retstatus = caam_jr_enqueue(&jobctx, NULL);
+
+	if (data->q && retstatus == CAAM_JOB_STATUS) {
+		/*
+		 * Expect to have a retstatus == CAAM_JOB_STATUS, where
+		 * job status == STATUS_GOOD_Q
+		 */
+		RSA_TRACE("Check Prime Q Status 0x%08" PRIx32, jobctx.status);
+
+		if (JRSTA_GET_HALT_USER(jobctx.status) == STATUS_GOOD_Q) {
+			cache_operation(TEE_CACHEINVALIDATE, data->p->data,
+					data->p->length);
+			cache_operation(TEE_CACHEINVALIDATE, data->q->data,
+					data->q->length);
+
+			RSA_DUMPBUF("Prime P", data->p->data, data->p->length);
+			RSA_DUMPBUF("Prime Q", data->q->data, data->q->length);
+			retstatus = CAAM_NO_ERROR;
+		}
+	} else if (retstatus == CAAM_NO_ERROR && !data->q) {
+		cache_operation(TEE_CACHEINVALIDATE, data->p->data,
+				data->p->length);
+
+		RSA_DUMPBUF("Prime", data->p->data, data->p->length);
+	}
+
+	if (retstatus != CAAM_NO_ERROR) {
+		RSA_TRACE("Prime Status 0x%08" PRIx32, jobctx.status);
+		retstatus = CAAM_FAILURE;
```

Comment:
Is there a good reason for overwriting the actual error code here? Aren't the callers expected to test for `!= CAAM_NO_ERROR` rather than `== CAAM_FAILURE` for instance?

Reply:
At some point, it was needed but not anymore. I will remove ```retstatus = CAAM_FAILURE;``` and change the ```if (retstatus != CAAM_NO_ERROR)``` into ```else if (retstatus != CAAM_NO_ERROR)```